fastlane/docs/Plugins.md

```diff
@@ -0,0 +1,149 @@
+# fastlane Plugins
+
+## Local actions
+
+You can create your own actions to extend the functionality of fastlane for your project. The action you create will behave exactly like the built in actions.
+
+Just run `fastlane new_action`, enter the name of the action and edit the generated Ruby file in `fastlane/actions/[action_name].rb`. After you finished writing your action, add it to your version control, so it is available for your whole team.
+
+From then on, you can just use your action in your `Fastfile`, just like any other action.
+
+### Submitting the action to the fastlane main repo
+
+Please be aware we may not accept all actions submitted to be bundled with `fastlane`. Before you submit a pull request adding the action to the `fastlane` code base, submit an issue proposing the new action and why it should be built-in.
+
+In general we tend to accept actions that
+
+- Generally usable for a big majority of developers (e.g. basic interactions with git)
+- Solve pain points for mobile app developers (iOS and Android)
+- have an easy to read documentation and great test coverage
+
+In general, we might not accept actions that
+
+- Solve specific use-cases for only a small subset of developers
+- Access the API of a third party service, the third party service should own and maintain the action
+- Complex actions, that will require a lot of work to maintain in the future
+- Everything that isn't mobile developer related
+
+## Remote plugins
+
+`fastlane` is an open platform and we enable every developer to extend it to fit their needs. That's why we built a plugin system that allows you and your company to provide `fastlane` plugins to other `fastlane` users. You have the full power and responsibility of maintaining your plugin and keeping it up to date. This is useful if you maintain your own library or web service, and want to make sure the `fastlane` plugin is always up to date.
+
+### Find a plugin
+
+Head over the [AvailablePlugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/AvailablePlugins.md) for a list of plugins you can use.
+
+List all available plugins using
+
+```
+fastlane search_plugins
+```
+
+To search for something specific 
+```
+fastlane search_plugins [query]
+```
+
+### Add a plugin to your project
+
+```
+fastlane add_plugin [name]
+```
+
+`fastlane` will assist you on setting up your project to start using plugins.
+
+This will:
+
+- Add the plugin to `fastlane/Pluginfile`
+- Make sure your `fastlane/Pluginfile` is properly referenced from your `./Gemfile`
+- Run `bundle install` to make sure all required dependencies are installed on your local machine (this step might ask for your admin password to install Ruby gems)
+
```

Comment:
However by doing this once, people are more likely to remember, so let's keep it there